openziti-init-entrypoint.sh

```diff
@@ -0,0 +1,80 @@
+  --primary-ext-jwt-allowed \
+  --primary-ext-jwt-allowed-signers "${ext_jwt_id}")
+
+function deleteEdgexService {
+  local svc="edgex.$1"
+  ziti edge delete config "${svc}-intercept"
+  ziti edge delete service "${svc}"
+  ziti edge delete erp "${svc}.erp"
+  ziti edge delete sp "${svc}.bind"
+  ziti edge delete sp "${svc}.dial"
+}
+
+function makeEdgexService {
+  local svc="edgex.$1"
+  local int="$1.edgex.ziti"
+  ziti edge create identity "${svc}" --external-id "${1}" -a "edgex,${svc}-servers" -P "${edgex_auth_policy}"
+  ziti edge create config "${svc}-intercept" intercept.v1 '{"protocols":["tcp"],"addresses":["'"${int}"'"], "portRanges":[{"low":80, "high":80}]}'
+  ziti edge create service "${svc}" -a "${svc}" --configs "${svc}-intercept"
+  ziti edge create edge-router-policy "${svc}.erp" --identity-roles "#${svc}" --edge-router-roles '#all'
+  ziti edge create service-policy "${svc}.bind" Bind --service-roles "#${svc}" --identity-roles "#${svc}-servers"
+  ziti edge create service-policy "${svc}.dial" Dial --service-roles "#${svc}" --identity-roles "#${svc}-clients"
+}
+
+sleep 1
+makeEdgexService 'core-command'
+makeEdgexService 'core-data'
+makeEdgexService 'core-metadata'
+makeEdgexService 'device-virtual'
+makeEdgexService 'device-rest'
+makeEdgexService 'rules-engine'
+makeEdgexService 'support-notifications'
+makeEdgexService 'support-scheduler'
+makeEdgexService 'sys-mgmt-agent'
+makeEdgexService 'ui'
+makeEdgexService 'app-rules-engine'
+
+
+ziti edge create identity edgexuser -P "${edgex_auth_policy}" --external-id edgexuser -a edgex.device-virtual-servers,edgex.core-command-clients,edgex.core-data-clients,edgex.core-metadata-clients,edgex.device-virtual-clients,edgex.rules-engine-clients,edgex.support-notifications-clients,edgex.support-scheduler-clients,edgex.sys-mgmt-agent-clients
+ziti edge update identity edgex.rules-engine -a edgex.rules-engine-servers,edgex.core-command-clients
+ziti edge update identity edgex.device-virtual -a edgex.device-virtual-servers,edgex.core-command-clients,edgex.core-data-clients,edgex.core-metadata-clients,edgex.device-virtual-clients,edgex.rules-engine-clients,edgex.support-notifications-clients,edgex.support-scheduler-clients,edgex.sys-mgmt-agent-clients
+ziti edge update identity edgex.ui -a edgex.device-virtual-servers,edgex.core-command-clients,edgex.core-data-clients,edgex.core-metadata-clients,edgex.device-virtual-clients,edgex.rules-engine-clients,edgex.support-notifications-clients,edgex.support-scheduler-clients,edgex.sys-mgmt-agent-clients
+ziti edge update identity edgex.device-rest -a edgex.device-rest-servers,edgex.core-command-clients,edgex.core-data-clients,edgex.core-metadata-clients,edgex.device-virtual-clients,edgex.rules-engine-clients,edgex.support-notifications-clients,edgex.support-scheduler-clients,edgex.sys-mgmt-agent-clients
+ziti edge update identity edgex.core-command -a edgex,edgex.core-command-servers,edgex.core-metadata-clients,edgex.device-virtual-clients
```

Comment:
Can these be part of makeEdgexService so we don't have to also repeat this for all possible edgex services.

Reply:
since each one has individual, and discrete needs, I wanted to highligh only allowing communication from the services that are expected to communicate with other services. It's always easier to "let all the services talk to all the services", but I opted to follow the principle of least privilege here. 

I'll accomodate EdgeX however is best

Reply:
Yes, I am ok if you want to do this for every edgex App and Device  service ...

Reply:
I can but, are there broad rules that are applicable to all "app-*" services? Like any app-service can talk to any core service? I need to know which services are authorized to talk to which services. I can't do that through trial and error.

Reply:
App Services can talk to any core/support service
Device Services will only talk to Core MetaData

Reply:
is it expected for core services to call other core services at all?

Reply:
Yes, Core Command calls on Core MetaData.

Also just remembered that Support schedule calls Core Data, but no thru the Core Data Clients.
See configuration here: https://github.com/edgexfoundry/edgex-go/blob/main/cmd/support-scheduler/res/configuration.yaml#L16-L20

The schedule can all on any endpoint of any service, not just EdgeX services. The configuration linked above just happens to be calling on Core Data. Do we simply need to env  override to host and port for this to work?

Reply:
So the only core service that needs to communicate to another core service is core-command -> core-metadata and support-scheduler -> core-data but through the underlay? If that behavior is desired (using the underlay) technically the healthcheck proxy could be used for that, but it feels wrong to do that. It'd be better to adapt `scrub-aged-events` to use OpenZiti in some way.

At this time, I opted for core services to be authorized to dial other core services but that's easy to fix. The UI for example needs to dial all the core services, the two support services, two device services and an app service...

Have a look at the new openziti-init-entrypoint.sh specifically the new service-policy creation:
```

ziti edge create service-policy app-core-dial Dial --identity-roles "#application.id" --service-roles "#core.svc"
ziti edge create service-policy app-support-dial Dial --identity-roles "#application.id" --service-roles "#support.svc"
ziti edge create service-policy ds-core-dial Dial --identity-roles "#device.id" --service-roles "@edgex.core-metadata"
ziti edge create service-policy core-core-dial Dial --identity-roles "#core.id" --service-roles "#core.svc"
```

* app identities authorized to dial core services
* app identities authorized to dial support services
* device identities authorized to dial core metadata only
* core identities authorized to core support services

Reply:
> At this time, I opted for core services to be authorized to dial other core services but that's easy to fix. The UI for example needs to dial all the core services, the two support services, two device services and an app service...

Actuall, I can't think of where the UI actually calls on the Device Services. If you found somewhere then it applies to all device service, not just the two in the current compose file.